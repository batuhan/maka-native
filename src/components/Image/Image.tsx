import React, { ReactNode } from "react";
import { Image } from "react-native";
import styled from "styled-components";

interface Props {
  style?: {};
  source: string;
}

export default function HeaderComponent({ source, style }: Props) {
  return <Image style={style} source={require(source)} />;
}
