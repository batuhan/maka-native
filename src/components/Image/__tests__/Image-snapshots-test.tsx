import React from "react";
import renderer from "react-test-renderer";

import Image from "../Image";

describe("Image", () => {
  it("renders correctly", () => {
    const source = "../../assets/images/image-test.jpg";
    const instance = renderer.create(<Image style={{}} source={source} />);
    expect(instance.toJSON()).toMatchSnapshot();
  });
});
