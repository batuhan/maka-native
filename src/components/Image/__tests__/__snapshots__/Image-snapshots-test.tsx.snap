// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders correctly 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/assets/images/image-test.jpg",
    }
  }
  style={Object {}}
/>
`;
